commit 8098ab85dc79c5d039404142c4182f341a23ac0c (refs/changes/15/3715/5)
Author: Mike Gerdts <mike.gerdts@joyent.com>
Date:   2018-04-03T16:12:50+00:00 (1 year, 6 months ago)
    
    OS-6799 hvm exclusion failures are silent
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Reviewed by: Jerry Jelinek <jerry.jelinek@joyent.com>
    Reviewed by: John Levon <john.levon@joyent.com>
    Approved by: Patrick Mooney <patrick.mooney@joyent.com>

diff --git a/usr/src/uts/i86pc/os/pc_hvm.c b/usr/src/uts/i86pc/os/pc_hvm.c
index b74ceb17bf..60f8e8d3ca 100644
--- a/usr/src/uts/i86pc/os/pc_hvm.c
+++ b/usr/src/uts/i86pc/os/pc_hvm.c
@@ -10,9 +10,12 @@
  */
 
 /*
- * Copyright 2017 Joyent, Inc.
+ * Copyright 2018 Joyent, Inc.
  */
 
+#include <sys/cmn_err.h>
+#include <sys/ddi.h>
+#include <sys/sunddi.h>
 #include <sys/param.h>
 #include <sys/types.h>
 #include <sys/mutex.h>
@@ -35,12 +38,17 @@ static const char *hvm_excl_holder = NULL;
 boolean_t
 hvm_excl_hold(const char *consumer)
 {
-	boolean_t res = B_FALSE;
+	boolean_t res;
 
 	mutex_enter(&hvm_excl_lock);
 	if (hvm_excl_holder == NULL) {
 		hvm_excl_holder = consumer;
 		res = B_TRUE;
+	} else {
+		cmn_err(CE_WARN, "zone '%s' cannot take HVM exclusion lock as "
+		    "'%s': held by '%s'", curproc->p_zone->zone_name, consumer,
+		    hvm_excl_holder);
+		res = B_FALSE;
 	}
 	mutex_exit(&hvm_excl_lock);
 
